base_layer/core/src/proof_of_work/difficulty.rs

```diff
@@ -83,6 +83,18 @@ impl Difficulty {
         let result = result.min(u64::MAX.into());
         Difficulty::from_u64(result.low_u64())
     }
+
+    pub fn checked_div_u64(&self, other: u64) -> Option<Difficulty> {
+        if other == 0 {
+            return None;
+        }
+        let result = Difficulty(self.0 / other);
+        if result.0 < MIN_DIFFICULTY {
+            None
+        } else {
+            Some(result)
+        }
+    }
```

Comment:
There is a `checked_div` function for u64 already. It does divide by 0 checking. Is there a reason not to use it?
https://doc.rust-lang.org/stable/std/primitive.u64.html#method.checked_div

Additionally, this function is doing more than a standard `checked_*` function. We're comparing it against our own min difficulty. So we could go for a name that doesn't match a pre-existing usage pattern.

```suggestion
    pub fn min_difficulty_checked_div(&self, other: u64) -> Option<Difficulty> {
        return match self.0.checked_div(other) {
            None => None,
            Some(n) => {
                n < MIN_DIFFICULTY ? None : Some(Difficulty(n))
            }
        }
    }
```

Reply:
Additionally, the `MIN_DIFFICULTY` constant is well... Constant. Shouldn't this be compared against the current minimum proof of work difficulty?

Is there a chance someone could use this to always achieve the minimum difficulty of 1? n could be the same as the constant which would result in a min difficulty lower than the pow difficulty.

Reply:
Min_difficulty is a constant, and that why we need to keep checking it against that. And the context here keeps it to difficulty, and does not go towards having a min_chain difficulty that dependant on previous timestamps etc